demo-apps/cf-android/app/src/main/java/org/eclipse/californium/examples/MainActivity.java

```diff
@@ -100,7 +153,23 @@ protected void onPostExecute(CoapResponse response) {
                 ((TextView)findViewById(R.id.textContent)).setText(response.getResponseText());
             } else {
                 ((TextView)findViewById(R.id.textCodeName)).setText("No response");
+                client.shutdown();
```

Comment:
Why do you want to call shutdown()? Does your intention match the javadoc definition of the function?